packages/@react-aria/table/docs/useTable.mdx

```diff
@@ -488,12 +492,823 @@ function Checkbox(props) {
   let ref = React.useRef();
   let state = useToggleState(props);
   let {inputProps} = useCheckbox(props, state, ref);
-  return <input {...inputProps} ref={ref} />;
+  return <input {...inputProps} ref={ref} style={props.style} />;
 }
 ```
 
 </details>
 
+## Resizable Columns
+
+For resizable column support, two additional hooks need to be added to the table implementation above. The <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTableColumnResizeState} />
+hook from `@react-stately/table` is responsible for initializing and tracking the widths of every column in your table, returning functions that you can use to
+update the column widths during a column resize operation. Note that this state is supplementary to the state returned by <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTableState} />.
+
+The second column resizing hook is <TypeLink links={docs.links} type={docs.exports.useTableColumnResize} />. This hook handles the interactions for the a table column's resizer
+element, allowing the user to drag the resizer or use the keyboard arrows to expand the column's width. Be sure to pass the state returned by <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTableColumnResizeState} />
+to this hook so the tracked widths can be updated appropriately. We'll walk through all the changes to the previous table implementation step by step below.
+
+### Table
+
+As mentioned previously, we first need to call <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTableColumnResizeState} /> to initialize and update the widths for our table's columns.
+By providing the hook with `getDefaultMinWidth` and `getDefaultWidth`, we can give our table selection checkbox column a pre-determined width while allowing the other columns divide up the remaining
+width equally initially. We'll pass the state returned by <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTableColumnResizeState} /> along with any user defined `onResize` handlers
+to our `ResizableTableColumnHeaders` so it can be used by <TypeLink links={docs.links} type={docs.exports.useTableColumnResize} />. The `widths` from this state are provided to each table element so
+the table cells in the body match the columns at all times.
+
+The various style changes below are to make the table itself scrollable and to support table body/column widths greater than the 400px applied to the table itself.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+/*- begin highlight -*/
+import {useTableColumnResizeState} from '@react-stately/table';
```

Comment:
Open to opinions. The changes made can be argued to be a bit extensive/excessive for an aria example but what I have here is what I felt to be a happy middle ground. Will be pointing out gotchas/places for alternative implementation here and there which I'd like opinions on